internal/model/log.go

```diff
@@ -0,0 +1,38 @@
+package model
+
+// At this limit, with one resource having a 120k log, render time was ~20ms and CPU usage was ~70% on an MBP.
+// 70% still isn't great when tilt doesn't really have any necessary work to do, but at least it's usable.
+// A render time of ~40ms was about when the interface started being noticeably laggy to me.
+const maxLogLengthInBytes = 120 * 1000
+
+type Log struct {
```

Comment:
i worry that the use of pointer receiver tricks to control mutability might be a bit too clever. maybe we should have two separate structs, Log and ReadOnlyLog?

Reply:
Maybe I misunderstand what I did here - how is this a "trick" or "clever" and not just the almost the most basic use of pointers?

The use of pointer receivers here is not to *control* mutability, but simply because we can't use pointer receivers throughout renderer.go because there's a bunch of code like `LastBuild().Log.String()`

Making these two separate structs is non-trivial since we need to be able to call both `String` and `Append` on `BuildRecord.Log` (though maybe we want to do that anyway since the renderer should be operating on an unchanging copy of the log, and it looks like right now it's sharing these bytes with the engine state).

I guess I don't really understand your concern since changing `Append` to be a function instead of a method would mean we're no longer using pointer receiver "tricks", but I don't see what problem it solves.